clusters/resource_cluster.go

```diff
@@ -140,6 +147,34 @@ func Validate(cluster any) error {
 // Long term, ModifyRequestOnInstancePool() in clusters_api.go will be removed once all the resources using clusters are migrated to Go SDK.
 func ModifyRequestOnInstancePool(cluster any) error {
 	switch c := cluster.(type) {
+	case *compute.ClusterSpec:
```

Comment:
should these supporting functions be split into a clusters_api_go_sdk.go file? I don't like the code duplication here, but this is more of an issue with the underlying sdk structs

Reply:
Yes, that is a good point. I can move them over in a later refactor PR given that this one is already really big